src/main.rs

```diff
@@ -936,7 +1011,7 @@ where
             });
         }
     }
-    if !is_whitelisted_da_receiver && *USE_FASTAUTH_FEATURES {
+    if !is_whitelisted_da_receiver && *USE_FASTAUTH_FEATURES && *USE_WHITELISTED_CONTRACTS  {
```

Comment:
USE_FASTAUTH_FEATURES should imply USE_WHITELISTED_CONTRACTS since fastauth always uses that